Linked list basic 리팩토링 했습니다

hiuk opened the pull request:
Linked list basic 리팩토링 했습니다.

hiuk committed:
Upload linked list (basic)

upload source file and input file.

kwangilcho commented on Linked list/linked_list_basic.c:
```diff
@@ -0,0 +1,170 @@
+#define MAX_BUF 128
+#define MAX_ARGUMENT 2
+
+struct Node{
```
```c
struct Node {
```

kwangilcho replied:
space

kwangilcho commented on Linked list/linked_list_basic.c:
```diff
@@ -0,0 +1,170 @@
+    struct Node *next;
+};
+
+struct List{
```
```c
struct List {
```

kwangilcho replied:
space

kwangilcho commented on Linked list/linked_list_basic.c:
```diff
@@ -0,0 +1,170 @@
+    struct Node *header;
+};
+
+enum Command{
```
```c
enum Command {
```

kwangilcho replied:
space

kwangilcho commented on Linked list/linked_list_basic.c:
```diff
@@ -0,0 +1,170 @@
+    while (pos) {
+        if (pos->element == element) {
+            break;
+        } else {
```
else가 없어도 괜찮겠죠?

kwangilcho commented on Linked list/linked_list_basic.c:
```diff
@@ -0,0 +1,170 @@
+    while (pos->next) {
+        if (pos->next->element == element) {
+            break;
+        } else {
```
else가 없어도 괜찮겠죠?

kwangilcho commented on Linked list/linked_list_basic.c:
```diff
@@ -0,0 +1,170 @@
+    return new_list;
+}
+
+int IsEmpty(struct Node *list)
```
```c
struct List
```
를 받아야하지 않을까?

kwangilcho commented on Linked list/linked_list_basic.c:
```diff
@@ -0,0 +1,170 @@
+
+struct Node *CreateNode(int element)
+{
+    struct Node *new_node = (struct Node *)malloc(sizeof(struct Node));
```
변수 이름에 타입을 지칭하는 단어를 굳이 쓸 필요가 없습니다.
C 언어는 정적 타입 언어이기 때문에 반드시 변수를 생성할 때 어떤 타입의 변수인지를 명시하게 되어있습니다.
즉, C로 코드를 작성하는 사람은 반드시 타입을 보게 되어있다는 뜻입니다.
그러므로 변수에 타입에 대한 정보를 담는 것 자체가 중복이라 볼 수 있습니다.

kwangilcho commented on Linked list/linked_list_basic.c:
```diff
@@ -0,0 +1,170 @@
+
+struct List *CreateList()
+{
+    struct List *new_list = (struct List *)malloc(sizeof(struct List));
```
변수 이름에 타입을 지칭하는 단어를 굳이 쓸 필요가 없습니다.
C 언어는 정적 타입 언어이기 때문에 반드시 변수를 생성할 때 어떤 타입의 변수인지를 명시하게 되어있습니다.
즉, C로 코드를 작성하는 사람은 반드시 타입을 보게 되어있다는 뜻입니다.
그러므로 변수에 타입에 대한 정보를 담는 것 자체가 중복이라 볼 수 있습니다.

kwangilcho reviewed (commented):
잘 구현하셨네요. 👍 
커밋 내용을 조금 더 상세하게 작성하면 더 좋을 것 같습니다.
피드백 남겼어요.

hiuk committed:
Modify linkedlist(basic) applying feedback

- Remove unnecessary spaces in structure and enmueraion definition.
- Remove unnecessary else-statement in Find and FindPrevious function.
- Change input type "struct Node *" to "struct List *" in IsEmpty
  function.
- Remove unnecessary type name from variables in CreateNode and CreateList
  function.
